server.js

```diff
@@ -14,10 +17,60 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+let messages = [welcomeMessage];
+
+app.get("/", function (req, res) {
+  res.sendFile(__dirname + "/index.html");
+  // res.json("I am here!");
+});
+
+// Search text in messages -  /messages/search?text=express
+app.get("/messages/search", (req, res) => {
+  const {text }= req.query;
+  const filteredMessages = messages.filter((message) =>
+   message.text.includes(text));
+  res.json(filteredMessages);
+});
+
+// get last 10 messages -  /messages/latest
+app.get("/messages/latest", (req, res) => {
+  res.json(messages.slice(-10));
+});
+
+
+// Read one message specified by an ID
+app.get("/messages/:id", (req, res) => {
+  const message = messages.find(
+    (message) => message.id === Number(req.params.id)
+  );
+  res.json(message);
+});
+
+//  Read all messages
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+// Delete a message, by ID
+app.delete("/messages/:id", (req, res) => {
+  messages = messages.filter((message) => message.id !== Number(req.params.id));
+  res.json(messages);
+});
+
+// Create a new message
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+  const newMessageObject = { id: messages.length, from, text };
+
+  if (from === undefined || from === "" || text === undefined || text === "") {
```

Comment:
Well done for validating the message! You could refactor your validation into a function to describe what it's doing, e.g. `validateMessage(newMessageObject)`